### PR TITLE
Fix: Collabocate client & API gives success response for access token's 401 error

### PR DESCRIPTION
**Pull request by KBandipo**
#### This pull request makes the following changes

* Fixes collabo-community/issue-tickets-ready-for-fixing#12

#

#### General checklist

- [x] My pull request adheres to the [Collabo Community pull request guidelines](https://docs.collabocommunity.com/pull-request-guidelines).
- [x] I have read and understood the **Pull request guideline extension** section found at the bottom of this pull request description/body.
- [x] I have replaced the dummy text under the subheading **Testing checklist** below with the testing checklist from the issue ticket I'm working on.

#

#### Testing checklist

- [x] API: The success message returned by the API has been replaced with Unauthorized: Can't access this resource.
- [x] Client's UI collects the Unauthorized: Can't access this resource message from the API, and uses it on the UI instead.
- [x] I certify that I ran my checklist

#

#### Pull request guideline extension

> [!NOTE]  
> Do not remove this note or section. This talks about extra things to take note of, in addition to the directions given in the Collabo Community pull request guidelines.

We appreciate the time taken to help tackle issue tickets within the Collabo Community. Please go ahead to submit your pull request even if it is not yet completed, and push to the pull request actively. This will allow us to know that you have not abandoned the issue ticket. However, you should only alert reviewers about your pull request as explained below, to avoid flooding their GitHub notifications unnecessarily.

"Ready for review" implies that things are working as expected, according to how it is specified in the issue ticket you are working on. Ensure that you test the code that you are working on, that it works as expected. Don't just request for review when your pull request is not yet done and working.

The only time you should be making people review something incomplete is: if you want to show them what's not working, so that they can send help or guidance. If this is the case, be explicit about this in our Discord community or in the comment of your pull request (after you have submitted the pull request).

### CODE REVIEW

**Comment by KBandipo** [comment]
Ready for review

**Comment by samuko-things** [comment]
> Ready for review

have you tested it on your end and is it working? @KBandipo

**Comment by KBandipo** [comment]
@samuko-things Yes, I have

**Review by Akpjunior94 — COMMENTED** [review]
@KBandipo Firstly By going through your code changes I can see that the second part of the expected behavior has not been implemented - the client UI error message. 
image

**Comment by Akpjunior94** [comment]
@KBandipo Do make sure you also check for the error message and display it on the Client's UI 
image

**Review by Akpjunior94 — APPROVED** [review]
Looks good to me.  @samuko-things you can give your Input on the backend API aspect part.

**Comment by samuko-things** [comment]
The backend-api looks good to me, I've gone through the changes made and tested it on my end.
good job @KBandipo.

My only problem is that the issue ticket and its title refer to the `403` (which is forbidden) error, but the checklist seems to address the UnAuthrized error (which is 401).
while testing on my end, it gives a `401` error, which @Akpjunior94 also got from the Discord discussion.

the PR addresses the `401` unauthorized error, which works, as expected based on the checklist.

In summary, I am okay with the PR, but I would only like @Ifycode to give her thoughts on this.
other than I'm good with it.

**Comment by Ifycode** [comment]
You can just change both the pull request title and the issue title to `401` instead of `403` @KBandipo 
After that you can merge it @Akpjunior94

**Comment by KBandipo** [comment]
@Ifycode I don't have access to edit the issue
